test/Schemes/SpecOpen.jl

```diff
@@ -25,9 +25,10 @@ end
   Y = subscheme(X, I)
   Xx = hypersurface_complement(X, x)
   Yx = hypersurface_complement(Y, x)
-  @test_broken complement(Y, Yx)
-  @test_broken is_closed_embedding(Y, Yx)
-  complement(X,Xx)
+  #@test_broken complement(Y, Yx) # Not a valid test! `complement` can only be carried out 
+                                  # whenever the first argument is closed in the second. 
```

Comment:
Then we should document that. 
Though I think that it makes sense to ask for the complement of an open set. That should return a closed set.
Note also that  we need the second argument closed in the first since we return `complement(X,V) = X \ V = X - V`

Reply:
Yes, I was thinking so, too. The problem is that `complement` is a rather generic function and one is not advised to document each of its methods. So I am not sure what is the best way to do that. Maybe a more informative error message is the way to go?

As for the complement of an open set: I disagree. The point is that the complement is not well defined as a scheme; it can have arbitrary multiplicities. For the type `PrincipalOpenSubset`, on the other hand, we have a known complement due to its history and in this case I would implement that function.

Reply:
Okay convinced. Well if our method has some special behavior in this case we should document it. 
For example there are many docstrings for `sub`. 
I think it is allowed to document the methods separately if the mathematics behind differs. 
Just don't document `Int` vs `fmpz` twice :-)